cms/djangoapps/api/v1/serializers/discussion_settings.py

```diff
@@ -0,0 +1,72 @@
+""" Discussion settings serializers. """
+
+
+from datetime import datetime
+
+from django.utils.dateparse import parse_date, parse_datetime
+from rest_framework import serializers
+
+from models.settings.course_metadata import CourseMetadata
+from xmodule.modulestore.django import modulestore
+
+
+def to_datetime(val, timepart):
+    """
+    Converts a given string to a datetime object. If the string only
+    contains date part, than additional timepart param used to build
+    datetime object. Raises ValueError if unable to parse given string.
+    """
+
+    parsed_val = parse_datetime(val)
+    if not parsed_val:
+        parsed_val = parse_date(val)
+        if parsed_val:
+            parsed_val = datetime.combine(parsed_val, timepart)
+        else:
+            raise ValueError('Invalid date')
+    return parsed_val
+
+
+def blackout_date_range_validator(value):
+    """
+    Given two date/datetime string, checks if those are valid dates
+    and if 2nd date/datetime is larger than 1st date/datetime. Doesn't
+    perform any transformation though, returns ``value`` as is if valid,
+    else raises ``serializers.ValidationError``.
+    """
+
+    [from_date, to_date] = value
+    try:
+        from_date = to_datetime(from_date, datetime.min.time())
+        to_date = to_datetime(to_date, datetime.max.time())
+        if from_date >= to_date:
+            raise serializers.ValidationError('Invalid Date Range')
+    except Exception:
+        raise serializers.ValidationError('Invalid Date')
+    return value
+
+
+class DiscussionSettingsSerializer(serializers.Serializer):  # pylint: disable=abstract-method
+    """
+    Serializer for discussion related settings. Also overrides ``save`` method
+    to support custom update flow of settings
+    """
+
+    allow_anonymous_to_peers = serializers.BooleanField()
+    allow_anonymous = serializers.BooleanField()
+    discussion_blackouts = serializers.ListField(
+        child=serializers.ListField(
+            child=serializers.CharField(),
```

Comment:
Perhaps this can validate as a datetime? If the input string doesn't have a "T" in it, then you append "T00:00" and parse it. Then the range validator can focus on validating the range only.

Reply:
As now we are using existing validation, I think a ``CharField`` is better. It keeps the old behaviour (both date & datetime are allowed). Also, existing validation already validates if given strings are date or datetime.